### PR TITLE
Connect API

### PR DESCRIPTION
**Pull request by roeierez**
This PR simplify the sdk initialization steps. Before the change the user had to execute different flows for `recover`, `register`, `init` which "leaked" the greenlight implementation details to the user.
The change aim to abstract the node initialization for these three use cases providing one method for all:

```rust
fn connect(
        config: Config,
        seed: Vec<u8>,
        event_listener: Box<dyn EventListener>)
```

The config now contains the "NodeConfig" which consist of the specific node type parameters, for greenlight it would be: 

```rust
pub struct GreenlightNodeConfig {
    pub partner_credentials: Option<GreenlightCredentials>,
    pub invite_code: Option<String>,
}
```

Currently we have one node provider but the NodeConfig was created as an enum to support additional node providers.
The initialization code looks the same for all use cases and doesn't require the user to understand greenlight's credentials internals, nor to save them:

```rust
let mut config = BreezServices::default_config(EnvironmentType.Production, "api key",
            NodeConfig::Greenlight {
                config: GreenlightNodeConfig {
                    partner_credentials: <partner credentials>,
                    invite_code: <invite code>,
                },
            },
);
let breez_services =
        BreezServices::connect(config, seed.to_vec(), Box::new(CliEventListener {})).await?;
let node_info breez_services.node_info()
```

Kotlin for example would look like this:

```kotlin
var config = breez_sdk.defaultConfig(breez_sdk.EnvironmentType.STAGING, "api key", breez_sdk.NodeConfig.Greenlight(breez_sdk.GreenlightNodeConfig(null, "")))
var sdkServices = breez_sdk.connect(config, seed, SDKListener());    
var nodeInfo = sdkServices.nodeInfo();
```

### CODE REVIEW

File: libs/sdk-core/src/greenlight/node_api.rs [hunk 1]
```diff
@@ -45,13 +48,104 @@ pub(crate) struct Greenlight {
...
+        let credentials = persister.get_gl_credentials()?;
+        let parsed_credentials: Result<GreenlightCredentials> = match credentials {
+            // In case we found existing credentials, try to decrypt them and connect to the node
+            Some(creds) => {
+                let decrypted_credentials = aes_decrypt(encryption_key_slice, creds.as_slice());
+                match decrypted_credentials {
+                    Some(creds) => {
+                        let built_credentials: GreenlightCredentials =
+                            serde_json::from_slice(creds.as_slice())?;
+                        info!("Initializing greenligihth from existing credentials");
```
Comment (dangeross) [inline libs/sdk-core/src/greenlight/node_api.rs:87]: nit: greenlight

File: tools/sdk-cli/src/command_handlers.rs [hunk 2]
```diff
@@ -69,59 +64,34 @@ pub(crate) async fn handle_command(
...
+                    partner_credentials,
+                    invite_code,
+                },
+            };
 
-            init_sdk(config, &persistence.get_or_create_seed(), &creds).await?;
-            persistence.save_credentials(creds)?;
-            Ok("Node was recovered succesfully".to_string())
+            connect(config, &persistence.get_or_create_seed()).await?;
+            Ok("Node was registered succesfully".to_string())
```
Comment (dangeross) [inline tools/sdk-cli/src/command_handlers.rs:93]: Node was connected successfully

File: tools/sdk-cli/src/command_handlers.rs [hunk 3]
```diff
@@ -223,8 +193,8 @@ pub(crate) async fn handle_command(
             let tx_ids = sdk()?.close_lsp_channels().await?;
             Ok(format!("Closing transaction ids:\n{:?}", tx_ids))
         }
-        Commands::StopNode {} => {
-            sdk()?.stop().await?;
+        Commands::Disconnect {} => {
+            sdk()?.disconnect().await?;
             Ok("Node was stopped succesfully".to_string())
```
Comment (dangeross) [inline tools/sdk-cli/src/command_handlers.rs:198]: s/succesfully/successfully

File: libs/sdk-bindings/tests/bindings/golang/test_breez_sdk.go [hunk 4]
```diff
@@ -26,22 +27,14 @@ func main() {
...
-		log.Fatalf("RecoverNode failed: %#v", err)
-	}
-
-	sdkServices, err := breez_sdk.InitServices(breez_sdk.DefaultConfig(breez_sdk.EnvironmentTypeStaging), seed, credentials, breezListener)
+	inviteCode := "code"
+	config := breez_sdk.DefaultConfig(breez_sdk.EnvironmentTypeStaging, "", breez_sdk.NodeConfigGreenlight{Config: breez_sdk.GreenlightNodeConfig{PartnerCredentials: nil, InviteCode: &inviteCode}})
+	sdkServices, err := breez_sdk.Connect(config, seed, breezListener)
 
 	if err != nil {
 		log.Fatalf("InitServices failed: %#v", err)
```
[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/tests/bindings/golang/test_breez_sdk.go]: Done

File: libs/sdk-bindings/tests/bindings/test_breez_sdk.cs [hunk 5]
```diff
@@ -5,11 +5,10 @@
 {
  var seed = BreezSdkMethods.MnemonicToSeed("cruise clever syrup coil cute execute laundry general cover prevent law sheriff");
  BreezSdkMethods.SetLogStream(new LogStreamListener());
- var credentials = BreezSdkMethods.RecoverNode(Network.BITCOIN, seed);
- BlockingBreezServices sdkServices = BreezSdkMethods.InitServices(BreezSdkMethods.DefaultConfig(EnvironmentType.STAGING), seed, credentials, new SDKListener());
- sdkServices.Start();
+ var config = BreezSdkMethods.DefaultConfig(EnvironmentType.STAGING, "", new NodeConfig.Greenlight(new GreenlightNodeConfig(null, "")));
```
[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/tests/bindings/test_breez_sdk.cs]: Done

File: libs/sdk-bindings/bindings-swift/Sources/BreezSDK/breez_sdkFFI.h
[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/bindings-swift/Sources/BreezSDK/breez_sdkFFI.h:1]: Done

File: libs/sdk-bindings/bindings-swift/Sources/BreezSDK/breez_sdk.swift
[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/bindings-swift/Sources/BreezSDK/breez_sdk.swift:1]: Done

File: libs/sdk-core/src/greenlight/node_api.rs (hunk 1 above)
Comment (roeierez) [reply libs/sdk-core/src/greenlight/node_api.rs]: Done

File: tools/sdk-cli/src/command_handlers.rs (hunk 2 above)
Comment (roeierez) [reply tools/sdk-cli/src/command_handlers.rs:93]: Done

File: tools/sdk-cli/src/command_handlers.rs (hunk 3 above)
Comment (roeierez) [reply tools/sdk-cli/src/command_handlers.rs]: Done

File: libs/sdk-bindings/bindings-swift/Sources/BreezSDK/breez_sdkFFI.modulemap
[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/bindings-swift/Sources/BreezSDK/breez_sdkFFI.modulemap:1]: Done

File: libs/sdk-bindings/tests/bindings/test_breez_sdk.py [hunk 6]
```diff
@@ -8,11 +8,9 @@ def on_event(self, event):
 
 def test():        
      seed = breez_sdk.mnemonic_to_seed("cruise clever syrup coil cute execute laundry general cover prevent law sheriff");
-     credentials = breez_sdk.recover_node(breez_sdk.Network.BITCOIN, seed)
-     sdk_services = breez_sdk.init_services(breez_sdk.default_config(breez_sdk.EnvironmentType.STAGING), seed, credentials, SDKListener())
-     sdk_services.start()
+     config = breez_sdk.default_config(breez_sdk.EnvironmentType.STAGING, "",  GreenlightNodeConfig("", ""))
```
[the remark this answers is not in the document]
Comment (roeierez) [reply libs/sdk-bindings/tests/bindings/test_breez_sdk.py]: Why?

[the remark this answers is not in the document]
Comment (dangeross) [reply libs/sdk-bindings/tests/bindings/test_breez_sdk.py]: Sorry, I added the comment before you fixed it

**Review by ok300 — APPROVED** [review]
LGTM

**Pull request merged**